cmd/config.go

```diff
@@ -29,26 +29,57 @@ const (
 
 // Furyconf is reponsible for the structure of the Furyfile
 type Furyconf struct {
-	VendorFolderName string         `yaml:"vendorFolderName"`
-	Versions         VersionPattern `yaml:"versions"`
-	Roles            []Package      `yaml:"roles"`
-	Modules          []Package      `yaml:"modules"`
-	Bases            []Package      `yaml:"bases"`
+	VendorFolderName string          `yaml:"vendorFolderName"`
+	Versions         VersionPattern  `yaml:"versions"`
+	Roles            []Package       `yaml:"roles"`
+	Modules          []Package       `yaml:"modules"`
+	Bases            []Package       `yaml:"bases"`
+	Provider         ProviderPattern `mapstructure:"provider"`
 }
 
-// Map from glob pattern to version associated (e.g. {"aws/*" : "v1.15.4-1"}
+// ProviderPattern is the abstraction of the following structure:
+//provider:
+//   modules:
+//     aws
+//      - uri: https://github.com/terraform-aws-modules
+//        label: ufficial-modules
+type ProviderPattern map[string]ProviderKind
+
+// ProviderKind is the abstraction of the following structure:
+//
+// modules:
+//   aws
+//    - uri: https://github.com/terraform-aws-modules
+//      label: ufficial-modules
+type ProviderKind map[string][]RegistrySpec
+
+//RegistrySpec contains the couple uri/label to identify each tf new repo declared
+type RegistrySpec struct {
+	BaseURI string `mapstructure:"url"`
+	Label   string `yaml:"label"`
```

Comment:
mixing yaml and mapstructure here? is it working fine?

Reply:
is working, probably should be better to put both as mapstructure? what do you think?